### PR TITLE
Update eclipse build files

### PR DESCRIPTION
**Pull request by kiritsuku**
This makes the Eclipse build files compatible with the latest version of Scala IDE 4.2. Since 2.12 support was added in Scala IDE 4.2, I don't think it is possible to get it up and running in a previous version of the IDE.

Review by @soc

### CODE REVIEW

**Comment by dragos** [comment]
LGTM

File: src/eclipse/README.md [hunk 1]
```diff
@@ -42,6 +42,13 @@ consider them unchanged:
 
         git update-index --no-assume-unchanged `find src/eclipse -iname .classpath -or -iname .project`
 
+0. The 2.1, sources of Scala need to be built with a 2.12 version of the compiler. One can configure a 2.12 Scala installation
```
Comment (soc) [inline src/eclipse/README.md]: > The 2.1,

Is this a typo?

**Comment by soc** [comment]
Apart from that, LGTM! Thanks a lot @sschaef!

File: src/eclipse/README.md (hunk 1 above)
Comment (kiritsuku) [inline src/eclipse/README.md]: yes

**Comment by lrytz** [comment]
thanks a lot!

**Pull request merged**